kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -14333,6 +14333,13 @@ public String expandMacros(String url) {
         try {
             ContentResource cr = getResource(ref);
 
+            long contentLength = cr.getContentLength();
+
+            if (contentLength > (10 * 1024 * 1024)) {
+                log.warn("{} is larger than 10MB, returning an empty Optional ...", ref);
+                return Optional.empty();
```

Comment:
Does returning an empty here result in some user facing message upstream? If so it looks good to me. 

I'd think the use would want to know that their file could not be opened and don't know if they need to see a specific reason for that.

Reply:
Done